mlflow/store/db_artifact_repo.py

```diff
@@ -0,0 +1,279 @@
+import logging
+import os
+from abc import ABCMeta
+import sqlalchemy
+import tempfile
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE, RESOURCE_DOES_NOT_EXIST
+
+from contextlib import contextmanager
+
+from mlflow.utils import extract_db_type_from_uri
+from mlflow.store.artifact_repo import ArtifactRepository
+from mlflow.utils.file_utils import relative_path_to_artifact_path
+
+from mlflow.exceptions import MlflowException
+from mlflow.store.dbmodels.initial_artifact_store_models import Base as InitialBase
+from mlflow.store.dbmodels.initial_artifact_store_models import SqlArtifact
+from mlflow.protos.databricks_pb2 import INTERNAL_ERROR
+from sqlalchemy import or_
+
+from six.moves import urllib
+
+_logger = logging.getLogger(__name__)
+
+
+def _relative_path(base_dir, subdir_path, path_module):
+    relative_path = path_module.relpath(subdir_path, base_dir)
+    return relative_path if relative_path is not '.' else None
+
+
+def _relative_path_local(base_dir, subdir_path):
+    rel_path = _relative_path(base_dir, subdir_path, os.path)
+    return relative_path_to_artifact_path(rel_path) if rel_path is not None else None
+
+
+def extract_db_uri_and_path(artifact_uri):
```

Comment:
Can you add unit tests for this method for multiple DB URIs with multiple schemes (`mssql` and `sqlite`), varying artifact path lengths, and additional URI components (dialects, drivers, params)? 

The existing unit tests only seem to address vanilla `sqlite://` URIs .